### PR TITLE
Use Default Store View Constant instead of '0'

### PR DESCRIPTION
**Pull request by edenduong**
### Description (*)

Should Use Default Store View Constant "\Magento\Store\Model\Store::DEFAULT_STORE_ID" instead of hard coded string '0'

### Manual testing scenarios (*)

1. Check the file changes of Source code
2. ->getProduct()->getStoreId() == '0' should replace by the constant : \Magento\Store\Model\Store::DEFAULT_STORE_ID

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Bundle/Block/Adminhtml/Catalog/Product/Edit/Tab/Bundle/Option.php [hunk 1]
```diff
@@ -7,6 +7,9 @@
 
 use Magento\Framework\Data\Form\Element\AbstractElement;
 
+/**
+ * Class \Magento\Bundle\Block\Adminhtml\Catalog\Product\Edit\Tab\Bundle\Option
```
Comment (Stepa4man) [inline app/code/Magento/Bundle/Block/Adminhtml/Catalog/Product/Edit/Tab/Bundle/Option.php:16]: We don't need such obvious info in doc block. There should be an informative description of what this class for

File: app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Tab/Options/Option.php [hunk 2]
```diff
@@ -14,6 +14,8 @@
 use Magento\Catalog\Api\Data\ProductCustomOptionInterface;
 
 /**
+ * Class \Magento\Catalog\Block\Adminhtml\Product\Edit\Tab\Options\Option
```
Comment (Stepa4man) [inline app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Tab/Options/Option.php:22]: We don't need such obvious info in doc block. There should be an informative description of what this class for

File: app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Tab/Options/Option.php [hunk 3]
```diff
@@ -272,10 +288,13 @@ public function getTemplatesHtml()
     }
 
     /**
+     * Get Option Values
+     *
      * @return \Magento\Framework\DataObject[]
      * @SuppressWarnings(PHPMD.CyclomaticComplexity)
      * @SuppressWarnings(PHPMD.NPathComplexity)
      * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
+     * phpcs:disable Generic.Metrics.NestingLevel
```
Comment (Stepa4man) [inline app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Tab/Options/Option.php]: Please get rid of phpcs suppressing. We shouldn't hide this problem.

File: app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Tab/Options/Option.php [hunk 4]
```diff
@@ -335,7 +354,7 @@ public function getOptionValues()
                             'sort_order' => $_value->getSortOrder(),
                         ];
 
-                        if ($this->getProduct()->getStoreId() != '0') {
+                        if ($this->getProduct()->getStoreId() != \Magento\Store\Model\Store::DEFAULT_STORE_ID) {
```
Comment (Stepa4man) [inline app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Tab/Options/Option.php]: Please import class name everywhere.

**Review by Stepa4man — CHANGES_REQUESTED** [review]
Hi @edenduong, nice to that you are still here and do a lot of jobs to make Magento better.
Thanks for your contribution. Please check out my comments.

**Comment by edenduong** [comment]
Hello @Stepa4man,

Thank you for your feedback :)
I have just changed the source code . 
And could you please continue to review the  https://github.com/magento/magento2/pull/23781 ?
Have a nice day!

**Review by Stepa4man — CHANGES_REQUESTED** [review]
Please add declare(strict_types=1) to all updated (and new) files
https://devdocs.magento.com/guides/v2.3/coding-standards/technical-guidelines.html#1-basic-programming-principles
It would be cool if you'll find time to read this [doc](https://devdocs.magento.com/guides/v2.3/coding-standards/technical-guidelines.html#1-basic-programming-principles). It's very usefull for contributors.

Could I also ask you to make the same changes at all your opened PRs?

Thanks in advance!

**Comment by edenduong** [comment]
@Stepa4man : Thank you for your feedback. I have just changed source code and other Pull Request.

**Review by Stepa4man — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @Stepa4man, thank you for the review.
[ENGCOM-5501](https://jira.corp.magento.com/browse/ENGCOM-5501) has been created to process this Pull Request
:eight_spoked_asterisk: @Stepa4man, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

File: app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Tab/Options/Option.php [hunk 5]
```diff
@@ -390,6 +371,63 @@ public function getOptionValues()
...
+    /**
+     * Get Option Value Of Group Select
+     *
+     * @param array $value
+     * @param \Magento\Catalog\Model\Product\Option $option
+     * @param boolean $showPrice
+     * @param int $scope
+     * @return array
+     */
+    public function getOptionValueOfGroupSelect($value, $option, $showPrice, $scope)
```
Comment (hostep) [inline app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Tab/Options/Option.php]: This method should be `private` instead of `public` I think?

Comment (edenduong) [reply app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Tab/Options/Option.php]: yes. I have just changed it. Thanks.

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed

**Pull request merged**